noir_stdlib/src/hash/poseidon/bn254/consts.nr

```diff
@@ -0,0 +1,179 @@
+// Constants for various Poseidon instances in the case of the prime field of the same order as BN254.
+// Consistent with https://github.com/iden3/circomlib/blob/master/circuits/poseidon.circom and https://github.com/iden3/circomlib/blob/master/circuits/poseidon_constants.circom
+
+// Number of full rounds
+global rf = 8;
+
+// Number of partial rounds
+fn rp() -> [u8; 16]
+{
+    [56, 57, 56, 60, 60, 63, 64, 63, 60, 66, 60, 65, 70, 60, 64, 68]
+}
+
+// S-box power
+global alpha = 5;
+
+// Number of partial rounds for state of size t is given by rp()[t-2]
+
+fn ark2() -> [Field; 128]
```

Comment:
I wonder if we should not have a struct to aggregate the parameters, then we would get them with one function param2(), instead of doing ark2() and mds2(), which will prevent errors of using ark2() and mds3(). The latter will be catched by the type system because of the sizes not matching, but it is less user-friendly.